src/cc/tools/qfsfileenum_main.cc

```diff
@@ -93,50 +100,77 @@ main(int argc, char **argv)
         cout << "qfs client failed to initialize\n";
         return 1;
     }
-
-    KfsClient::BlockInfos infos;
-    retval = kfsClient->EnumerateBlocks(filename, infos);
-    if (retval < 0) {
-        cout << filename << ": " << ErrorCodeToStr(retval) << "\n";
-    } else {
-        int64_t space = 0;
-        cout << "block count: " << infos.size() << " blocks:\n";
-        const KfsClient::BlockInfo* pbi = 0;
-        for (KfsClient::BlockInfos::const_iterator it = infos.begin();
-                it != infos.end();
-                ++it) {
-            KfsClient::BlockInfo bi = *it;
-            cout <<
-                "position: "  << setw(10) << bi.offset  << setw(0) <<
-                " id: "       << setw(10) << bi.id      << setw(0) <<
-                " version: "  << setw(3)  << bi.version << setw(0) <<
-                " size: "     << setw(8)  << bi.size    << setw(0) <<
-                " location: " << setw(12) << bi.server  << setw(0)
-            ;
-            if (pbi && pbi->id == bi.id) {
-                if (pbi->offset != bi.offset) {
-                    cout << " *position mismatch*";
-                    retval = -EINVAL;
+    int i = optind;
+    if (! filename && i < argc) {
+        filename = argv[i++];
+    }
+    string msg;
+    while (filename) {
+        KfsClient::BlockInfos infos;
+        int status = kfsClient->EnumerateBlocks(filename, infos);
+        if (status < 0) {
+            cout << filename << ": " << ErrorCodeToStr(status) << "\n";
```

Comment:
endl manipulator flushes output stream. No reason to flush output stream in this case. Typically endl is wildly misused, possibly due to misconception that it does "line ending conversion", which it does not do.